src/linker/Linker/MemberReferenceExtensions.cs

```diff
@@ -13,8 +15,15 @@ public static string GetDisplayName (this MemberReference member)
 			case MethodReference method:
 				return method.GetDisplayName ();
 
+			case IMemberDefinition memberDef:
+				var sb = new StringBuilder ();
+				if (memberDef.DeclaringType != null)
+					sb.Append (memberDef.DeclaringType.GetDisplayName ()).Append ('.');
+				sb.Append (memberDef.Name);
+				return sb.ToString ();
+
 			default:
-				return member.FullName;
+				throw new NotImplementedException ();
```

Comment:
I would probably still return `FullName` in this case (and assert).
Any exception from the linker is fatal to the publish - so unless we really can't figure out what to do, we should not hard fail. In this case the only failure is to show a nice name - so definitely not serious enough to fail the publish.